app/lib/features/tasks/dialogs/create_task_list_sheet.dart

```diff
@@ -0,0 +1,169 @@
+    final titleInput = ref.watch(titleProvider);
+    final titleNotifier = ref.watch(titleProvider.notifier);
+    final textNotifier = ref.watch(textProvider.notifier);
+
+    return SideSheet(
+      header: 'Create new task list',
+      addActions: true,
+      body: Padding(
+        padding: const EdgeInsets.symmetric(horizontal: 15, vertical: 10),
+        child: Form(
+          key: _formKey,
+          child: Column(
+            crossAxisAlignment: CrossAxisAlignment.start,
+            children: <Widget>[
+              Padding(
+                padding: const EdgeInsets.symmetric(vertical: 10),
+                child: Row(
+                  children: [
+                    Expanded(
+                      child: SizedBox(
+                        height: 52,
+                        child: TextFormField(
+                          decoration: InputDecoration(
+                            hintText: 'Task list name',
+                            labelText: 'Name',
+                            border: OutlineInputBorder(
+                              borderRadius: BorderRadius.circular(5),
+                            ),
+                          ),
+                          controller: _titleController,
+                          onChanged: (String? value) {
+                            titleNotifier.state = value ?? '';
+                          },
+                          validator: (value) =>
+                              (value != null && value.isNotEmpty)
+                                  ? null
+                                  : 'Please enter a name',
+                        ),
+                      ),
+                    ),
+                  ],
+                ),
+              ),
+              MdEditorWithPreview(
+                onChanged: (String? value) {
+                  textNotifier.state = value ?? '';
+                },
+              ),
+              const SelectSpaceFormField(canCheck: 'CanPostTaskList'),
+            ],
+          ),
+        ),
+      ),
+      actions: [
+        ElevatedButton(
+          onPressed: () => context.canPop()
+              ? context.pop()
+              : context.goNamed(Routes.main.name),
+          style: ElevatedButton.styleFrom(
+            shape: RoundedRectangleBorder(
+              borderRadius: BorderRadius.circular(6),
+            ),
+            foregroundColor: Theme.of(context).colorScheme.neutral6,
+            textStyle: Theme.of(context).textTheme.bodySmall,
+          ),
+          child: const Text('Cancel'),
+        ),
+        const SizedBox(width: 10),
+        ElevatedButton(
+          onPressed: () async {
+            if (_formKey.currentState!.validate()) {
+              DefaultDialog(
+                title: Text(
+                  'Posting TaskList',
+                  style: Theme.of(context).textTheme.titleSmall,
+                ),
+                isLoader: true,
+              );
+              try {
```

Comment:
Can we have this non UI logic be a part of function which can be called here? Makes widget tree a lot clean and easier to refactor.